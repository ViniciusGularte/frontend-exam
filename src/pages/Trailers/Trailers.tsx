import React, { useContext, useState, useEffect } from "react";
import {
  Container,
  Sidebar,
  PlayListContainer,
  PlayListContainerItens,
  ContainerButtonMobile,
} from "./style";
import Logo from "../../components/Common/Logo/Logo";
import Button from "../../components/Common/Button/Button";
import VideoBlock from "../../components/TrailersList/VideoBlock";
import VideoApi from "../../service/route";
import { AppContext } from "../../store/context";
import { TrailerInterface } from "../../types/trailer";
function Trailers() {
  useEffect(() => {
    getTrailers();
  }, []);
  const [error, setError] = useState(false);
  const [trailers, setTrailers] = useState<TrailerInterface[]>([]);
  const { setLogin, setPassword, setEmail } = useContext(AppContext);

  //control side bar on mobile
  const [sidebar, setSidebar] = useState(false);
  const toogleSideBar = () => {
    setSidebar(!sidebar);
  };
  const getTrailers = async () => {
    const trailers = await VideoApi();
    console.log(trailers);
    if (trailers.error) {
    } else {
      setTrailers(trailers.items);
    }
  };
  const Logout = () => {
    setEmail("");
    setPassword("");
    setLogin(false);
  };
  return (
    <Container>
      <Sidebar sidebarStatus={sidebar}>
        <Logo />
        <Button
          longButton
          onClick={toogleSideBar}
          disable={sidebar ? false : true}
          label="TRAILERS"
        />
        <Button onClick={Logout} noBorder label="LOGOUT" />
      </Sidebar>
      <PlayListContainer sidebarStatus={sidebar}>
        <ContainerButtonMobile>
          <Button
            onClick={toogleSideBar}
            onlyMobile
            noMargin
            longButton
            label="MENU"
          />
        </ContainerButtonMobile>
        <PlayListContainerItens>
          {trailers.map((trailer: TrailerInterface, index: number) => (
            <VideoBlock
              thumbnail={trailer.snippet.thumbnails.standard.url}
              title={trailer.snippet.title}
            />
          ))}
        </PlayListContainerItens>
      </PlayListContainer>
    </Container>
  );
}
export default Trailers;
