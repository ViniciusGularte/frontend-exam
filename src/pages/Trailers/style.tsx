import styled from "styled-components";
import { devicesMax } from "../../assets/global_style";
const Container = styled.div`
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  height: 100vh;
`;
const Sidebar = styled.div`
  align-items: center;
  display: flex;
  flex: ${(props) => (props.sidebarStatus ? "100%" : "33%")};
  flex-direction: column;
  border-right: 1px solid #4c4c4c;
  height: 100%;
  @media ${devicesMax.mobileL} {
    display: ${(props) => (props.sidebarStatus ? "flex" : "none")};
  }
`;
const PlayListContainer = styled.div`
  display: flex;
  flex: 75%;
  align-items: center;
  background: rgb(0, 0, 1);
  background: linear-gradient(
    90deg,
    rgba(0, 0, 1, 1) 0%,
    rgba(0, 0, 0, 0.5284488795518207) 30%
  );

  height: 100%;
  width: 100%;
  flex-direction: column;

  @media ${devicesMax.mobileL} {
    display: ${(props) => (props.sidebarStatus ? "none" : "flex")};
    flex: 100%;
    background: linear-gradient(
      90deg,
      rgba(0, 0, 1, 1) 0%,
      rgba(0, 0, 0, 0.5284488795518207) 0%
    );
  }
`;
const PlayListContainerItens = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  overflow-y: auto;
`;
const ContainerButtonMobile = styled.div`
  display: flex;
  align-items: flex-end;
  margin: 16px;
`;
const ContainerErrorMessage = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 16px;
  p {
    color: ${(props) => props.theme.primary};
  }
`;
const ContainerLoadMore = styled.div`
  display: flex;
  flex: 1 0 40%;
  flex-direction: column;
  margin-left: 40px;
  margin-right: 40px;
  margin-top: 160px;
  align-items: center;
`;

export {
  Container,
  Sidebar,
  PlayListContainer,
  PlayListContainerItens,
  ContainerButtonMobile,
  ContainerErrorMessage,
  ContainerLoadMore,
};
