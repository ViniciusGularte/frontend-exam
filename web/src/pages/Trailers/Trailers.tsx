import React, { useContext, useState, useEffect } from "react";
import {
  Container,
  Sidebar,
  PlayListContainer,
  PlayListContainerItens,
  ContainerButtonMobile,
} from "./style";
import Logo from "../../components/Common/Logo/Logo";
import Button from "../../components/Common/Button/Button";
import VideoBlock from "../../components/TrailersList/VideoBlock";
import VideoApi from "../../service/route";
import { AppContext } from "../../store/context";
function Trailers() {
  useEffect(() => {
    getTrailers();
  }, []);
  const [error, setError] = useState(false);
  const [trailers, setTrailers] = useState([]);
  const { setLogin, setPassword, setEmail } = useContext(AppContext);

  //control side bar on mobile
  const [sidebar, setSidebar] = useState(false);
  const toogleSideBar = () => {
    setSidebar(!sidebar);
  };
  const getTrailers = async () => {
    const trailers = await VideoApi();
    console.log(trailers);
  };
  const Logout = () => {
    setEmail("");
    setPassword("");
    setLogin(false);
  };
  return (
    <Container>
      <Sidebar sidebarStatus={sidebar}>
        <Logo />
        <Button
          longButton
          onClick={toogleSideBar}
          disable={sidebar ? false : true}
          label="TRAILERS"
        />
        <Button onClick={Logout} noBorder label="LOGOUT" />
      </Sidebar>
      <PlayListContainer sidebarStatus={sidebar}>
        <ContainerButtonMobile>
          <Button
            onClick={toogleSideBar}
            onlyMobile
            noMargin
            longButton
            label="MENU"
          />
        </ContainerButtonMobile>
        <PlayListContainerItens>
          <VideoBlock />
          <VideoBlock />
          <VideoBlock />
          <VideoBlock />
          <VideoBlock />
          <VideoBlock />
        </PlayListContainerItens>
      </PlayListContainer>
    </Container>
  );
}
export default Trailers;
