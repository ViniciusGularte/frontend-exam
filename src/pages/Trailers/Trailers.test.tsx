import Enzyme, { mount } from "enzyme";
import ReactSeventeenAdapter from "@wojtekmaj/enzyme-adapter-react-17";
import Trailers from "./Trailers";

Enzyme.configure({ adapter: new ReactSeventeenAdapter() });
const mockCallBack = jest.fn();

describe("Trailers", () => {
  it("mount correctly", () => {
    const wrapper = mount(<Trailers />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button")).toHaveLength(4);
    expect(wrapper.find("button").at(0).text()).toEqual("CLOSE");
    expect(wrapper.find("button").at(1).text()).toEqual("TRAILERS");
    expect(wrapper.find("button").at(2).text()).toEqual("LOGOUT");
    expect(wrapper.find("button").at(3).text()).toEqual("MENU");
  });
});
