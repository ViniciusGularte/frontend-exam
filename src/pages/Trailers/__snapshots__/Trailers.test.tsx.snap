// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trailers mount correctly 1`] = `ReactWrapper {}`;
